includes/Integrations/Jetpack.php

```diff
@@ -68,6 +83,76 @@ public function add_to_jetpack_sitemap( $post_types ) {
 		return $post_types;
 	}
 
+	/**
+	 * Add VideoPress to allowed mime types. If the site does not support videopress, this allow will be filtered out.
+	 *
+	 * @since 1.8.0
+	 *
+	 * @param array $mime_types Associative array of allowed mime types per media type (image, audio, video).
+	 *
+	 * @return array
+	 */
+	public function add_videopress( array $mime_types ) {
+		$mime_types['video'][] = self::VIDEOPRESS_MIME_TYPE;
+
+		return $mime_types;
+	}
+
+	/**
+	 * Filter admin ajax responses to change video/videopress back to mp4.
+	 *
+	 * @since 1.8.0
+	 *
+	 * @param array   $response   Array of prepared attachment data. @see wp_prepare_attachment_for_js().
+	 * @param WP_Post $attachment Attachment object.
+	 *
+	 * @return array
+	 */
+	public function filter_admin_ajax_response( array $response, WP_Post $attachment ) {
+		if ( self::VIDEOPRESS_MIME_TYPE !== $attachment->post_mime_type ) {
+			return $response;
+		}
+
+		// Get data from post, so we filter out requests from other locations.
+		$post_data = $_POST; // phpcs:ignore WordPress.Security.NonceVerification.Missing
+		if ( ! isset( $post_data['action'] ) || 'query-attachments' !== $post_data['action'] ) {
+			return $response;
+		}
+
+		if ( ! isset( $post_data['query'] ) && ! isset( $post_data['query']['source'] ) || 'web_stories_editor' !== $post_data['query']['source'] ) {
```

Comment:
`isset()` can be simplified like this:

```php
! isset( $post_data['query']['source'] )
```

Reply:
```suggestion
		if ( ! isset( $post_data['query']['source'] ) || 'web_stories_editor' !== $post_data['query']['source'] ) {
```

Reply:
```suggestion
		if ( ! isset( $post_data['query'], $post_data['query']['source'] ) || 'web_stories_editor' !== $post_data['query']['source'] ) {
```

Or maybe even.

Reply:
No, that's not needed. `isset( $post_data['query']['source'] )` handles both cases without errors or notices.